### PR TITLE
add nd2

### PR DESCRIPTION
**Pull request by tlambert03**
nd2 (nikon elements) file reader

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tlambert03** [comment]
i _think_ this is ready for review @conda-forge/help-python, @conda-forge/help-python-c
In order to test the finished build on linux, I used centos7... the plain `linux_64` still fails but I think that's ok right?  input welcome.

**Comment by tlambert03** [comment]
bump @@conda-forge/help-python

File: recipes/nd2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    - sysroot_linux-64 2.17  # [linux64]
+  host:
+    - python
+    - pip
+    - cython
+    - numpy
+    - libtiff
+    - libzlib
+    - jpeg
+    - _openmp_mutex  # [linux]
```
Comment (isuruf) [inline recipes/nd2/meta.yaml:29]: See conda-forge docs on how to specify openmp

**Review by isuruf — CHANGES_REQUESTED** [review]
Please don't change files outside of `recipes/nd2`

**Comment by tlambert03** [comment]
> Please don't change files outside of `recipes/nd2`

thanks @isuruf... still working on this one.  will clean it up before requesting review

**Comment by tlambert03** [comment]
hmm... the windows build is no longer succeeding (it was about a month ago).  Now seeing unresolved symbols:
```
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX64\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:src\sdk\Windows\x86_64\lib /LIBPATH:C:\bld\nd2_1636754089901\_h_env\libs /LIBPATH:C:\bld\nd2_1636754089901\_h_env\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\ATLMFC\lib\x64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\um\x64" /LIBPATH:C:\bld\nd2_1636754089901\_h_env\Library\lib /LIBPATH:C:\bld\nd2_1636754089901\_h_env\Library\lib nd2readsdk-static.lib /EXPORT:PyInit_latest build\temp.win-amd64-3.8\Release\src/nd2/_sdk/latest.obj src\sdk\Windows\x86_64\lib\jbig.lib src\sdk\Windows\x86_64\lib\libjpeg.lib src\sdk\Windows\x86_64\lib\limfile.lib src\sdk\Windows\x86_64\lib\lzma.lib src\sdk\Windows\x86_64\lib\nd2readsdk-static.lib src\sdk\Windows\x86_64\lib\tiff.lib src\sdk\Windows\x86_64\lib\tiffxx.lib src\sdk\Windows\x86_64\lib\zlib.lib /OUT:build\lib.win-amd64-3.8\nd2\_sdk\latest.cp38-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.8\Release\src/nd2/_sdk\latest.cp38-win_amd64.lib
     Creating library build\temp.win-amd64-3.8\Release\src/nd2/_sdk\latest.cp38-win_amd64.lib and object build\temp.win-amd64-3.8\Release\src/nd2/_sdk\latest.cp38-win_amd64.exp
  limfile.lib(JsonFileDeviceImpl.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4
  limfile.lib(IoMemoryBufferImpl.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4
  limfile.lib(JsonFileDevice.obj) : error LNK2001: unresolved external symbol __CxxFrameHandler4
```

 I believe static libs I'm linking against require `msvcp140.dll` ... and i guess that means I need vs2015 here? 
I'm aware that vs2015 is deprecated (and perhaps completely unavailable?) ... any suggestions, preferably short of requesting recompiled libraries?

**Comment by tlambert03** [comment]
welp, I think this is stuck again.  it's working on mac and linux, but not windows anymore.  I believe (but am not 100% sure) that I need to compile using vs2015 so that I can link the static pre-compiled objects (based on all the errors mentioned above: `error LNK2001: unresolved external symbol __CxxFrameHandler4`).  yes, I understand this is because i'm dealing with precompiled libs... but i'm afraid that's the situation we find ourselves in.

I'm not sure why it worked a month ago on windows... and can no longer access those build logs to check.  (fwiw, It also seems to work fine on `cibuildwheel`):

I tried:
- adding c_compiler `vs2015` to `conda_build_config.yaml` ... but I get an [unsatisfiable dependency](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=411120&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=06d62fd2-ab49-5add-9ee5-83b6882d7d2b&l=649) from conda_build
- adding `vc =14` at various places in the recipe (didn't change anything)

any tips @isuruf, @jaimergp

**Comment by tlambert03** [comment]
I've decided not to build for windows for now.  This is ready for review, working on linux (centos7) and mac is enough for me.  Ready for re-review @isuruf, @conda-forge/help-python-c

**Comment by jaimergp** [comment]
No clue either @tlambert03, we can always enable Windows on the feedstock later.

@isuruf I think the [licensing](https://github.com/tlambert03/nd2/blob/main/LICENSE) / [vendoring](https://github.com/tlambert03/nd2/blob/main/src/sdk/LICENSE) issues are covered now. There's a [detailed analysis of the symbols](https://github.com/tlambert03/nd2/issues/30) in the source repo as well if you need to check something else about these libs.

File: recipes/nd2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  requires:
+    - pip
+  commands:
+    - pip check
+
+about:
+  home: https://github.com/tlambert03/nd2
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE
```
Comment (jaimergp) [inline recipes/nd2/meta.yaml:54]: We need to include the sdk license here too. This field can be a list of paths so, something like this:

```suggestion
  license_file: 
    - LICENSE
    - src/sdk/LICENSE
```

I don't know if the `license` and `license_family` fields need to be updated to reflect this too.

**Comment by tlambert03** [comment]
thanks again for all your help @jaimergp!  much appreciated

**Comment by tlambert03** [comment]
>  but I think we'd need permission for conda-forge as an entity. Is it possible to ask the authors for a more specific language when it comes to redistribution?

Hi @jaimergp, @mbargull @chrisburr, @goanpeca:

I contact the upstream company and requested explicit permission for conda-forge and pypi to re-distribute binaries.  They sent me an updated email, when I have included in the repo [here](https://github.com/tlambert03/nd2/blob/main/src/sdk/LICENSE) and also included in the sdist.  I cut a new release and am using it here.

Looks like builds are passing.  It would be great to finally have this merged!  Thanks for all your help.

File: recipes/nd2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    - imagecodecs._jpeg2k
+  requires:
+    - pip
+  commands:
+    - pip check
+
+about:
+  home: https://github.com/tlambert03/nd2
+  license: BSD-3-Clause
+  license_family: BSD
```
Comment (jaimergp) [inline recipes/nd2/meta.yaml:54]:
```suggestion
  license_family: BSD AND LicenseRef-LIMnd2sdk
```

File: recipes/nd2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    - imagecodecs._jpeg2k
+  requires:
+    - pip
+  commands:
+    - pip check
+
+about:
+  home: https://github.com/tlambert03/nd2
+  license: BSD-3-Clause
+  license_family: BSD AND LicenseRef-LIMnd2sdk
```
Comment (jaimergp) [inline recipes/nd2/meta.yaml:54]:
```suggestion
  license: BSD-3-Clause AND LicenseRef-LIMnd2sdk
```

**Comment by tlambert03** [comment]
thank you @jaimergp !!

**Pull request merged**